pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -218,15 +102,11 @@ func (n *NetworkPolicyController) updateNamespace(oldObj, curObj interface{}) {
 	defer n.heartbeat("updateNamespace")
 	oldNamespace, curNamespace := oldObj.(*v1.Namespace), curObj.(*v1.Namespace)
 	klog.V(2).Infof("Processing Namespace %s UPDATE event, labels: %v", curNamespace.Name, curNamespace.Labels)
-	oldLabelSet, curLabelSet := labels.Set(oldNamespace.Labels), labels.Set(curNamespace.Labels)
-	affectedACNPsByOldLabels := n.filterPerNamespaceRuleACNPsByNSLabels(oldLabelSet)
-	affectedACNPsByCurLabels := n.filterPerNamespaceRuleACNPsByNSLabels(curLabelSet)
-	affectedACNPs := utilsets.SymmetricDifferenceString(affectedACNPsByOldLabels, affectedACNPsByCurLabels)
-	for cnpName := range affectedACNPs {
-		if cnp, err := n.cnpLister.Get(cnpName); err == nil {
-			n.reprocessCNP(cnp, false)
-		}
+	// No ClusterNetworkPolicies are affected if the Namespace's labels do not change.
+	if labels.Equals(oldNamespace.Labels, curNamespace.Labels) {
+		return
 	}
+	n.enqueueCNPsWithPerNamespaceRule()
```

Comment:
I was trying to understand why `PerNamespaceSelectors` and all the related stuff around it is not needed anymore. Is it just because we are simplifying the logic and instead of enqueuing a smaller set of CNPs based on which Namespace labels have changed, we enqueue all CNPs with per-Namespace rules? Is it something worth mentioning in the commit message / PR description?

Reply:
Yes, we cannot use per-Namespace index of internalNetworkPolicy because there could be a race condition:
1. AddCNP triggers syncInternalNetworkPolicy
2. syncInternalNetworkPolicy calls getAffectedNamespacesForAppliedTo, in which namespaces that match selfNamespace selectors are retrieved, it doesn't commit the internalNetworkPolicy to storage yet
3. AddNamespace doesn't find the above internalNetworkPolicy, so it doesn't trigger syncInternalNetworkPolicy
4. following step 2, syncInternalNetworkPolicy commits the internalNetworkPolicy to storage, it will be in a stale state

We should use per-Namespace index of ClusterNetworkPolicy, in same situation:
1. AddCNP triggers syncInternalNetworkPolicy
2. syncInternalNetworkPolicy calls getAffectedNamespacesForAppliedTo, in which namespaces that match selfNamespace selectors are retrieved, it doesn't commit the internalNetworkPolicy to storage yet
3. AddNamespace calls finds the ClusterNetworkPolicy, it trigger syncInternalNetworkPolicy
4. following step 2, syncInternalNetworkPolicy commits the internalNetworkPolicy to storage, it will be in a stale state
5. Because step 3 triggerred another syncInternalNetworkPolicy, this time it will count the new Namespace in.

I thought there shouldn't many per-Namespace rules when making the above change, but I agree it's probably an irrelavant change, I will continue filtering a smaller set in this PR.